### PR TITLE
Stable8 backport 15489

### PR DESCRIPTION
**Pull request by MorrisJobke**
Backport of #15489 

cc @butonic @blizzz @nickvergessen

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Steps to test this: https://github.com/owncloud/core/pull/15332#issuecomment-88530663

**Comment by DeepDiver1975** [comment]
@MorrisJobke unit tests are broken 

	Test Result (4 failures / +4)

    Test\Preview.testIsMaxSizeWorking
    OCA\Files\TagServiceTest.testUpdateFileTags
    OCA\Files_Encryption\Tests\Hooks.testSetPassphrase
    OCA\Files_Encryption\Tests\Util.testFilterShareReadyUsers

**Comment by MorrisJobke** [comment]
@DeepDiver1975 I'm already looking into this :)

**Comment by karlitschek** [comment]
nice :+1:

**Comment by karlitschek** [comment]
would be nice to have thus in today

**Comment by blizzz** [comment]
@MorrisJobke one sharing test is still failing

**Comment by MorrisJobke** [comment]
> would be nice to have thus in today

I would like to have this tested more intense, because the code between stable7 and stable8 changed a lot and also between stable8 and master where some changes. This is the sum of both PRs. I'm against having this in that short before a release - even if it fixes stuff that happened, but 8.0.4 shouldn't be that far away ;)

**Comment by MorrisJobke** [comment]
@schiesbn Can you have a look at the failing unit tests. It seems that it can decrypt the key even if the password has changed recently. Has something changed lately to behave a little weird?

**Comment by DeepDiver1975** [comment]
> would be nice to have thus in today

@karlitschek where is the urgent need to merge asap coming from? We are still in freeze for 8.0.3.

**Comment by karlitschek** [comment]
@DeepDiver1975 well only the well known customer issues. I would hate to delay this for the next release

**Comment by DeepDiver1975** [comment]
Wow - seriously this change holds 12 commits containing +373 −142 SLOC of change. And unit tests are still failing. No-go from my POV.



> well only the well known customer issues.

Individual patches can be supplied via GCX from my understanding

**Comment by MorrisJobke** [comment]
Unit test should now pass. It was a little typo. :see_no_evil:

**Comment by butonic** [comment]
these changes are in oc7 and master. strictly speaking I have to agree with @DeepDiver1975. But I want this in as well... I promise to spend most of tomorrow testing. 

download

**Comment by DeepDiver1975** [comment]
> But I want this in as well... I promise to spend most of tomorrow testing.

This will delay 8.0.3 by just another weeks then because QA has to be redone @jnfrmarks 

Seriously we have a process and workflow for a reason. There are a hell of people out there waiting for 8.0.3 for more then 3 weeks. We did revert other prs (with far less risk impact) last week.

Just restarting this discussion on every release we do is so Kindergarden.

**Comment by jnfrmarks** [comment]
Is this being added to 8.0.3?

**Comment by MorrisJobke** [comment]
> Is this being added to 8.0.3?

No.

**Comment by MorrisJobke** [comment]
@butonic @blizzz Please review this

File: lib/private/user/nouserexception.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING-AGPL file.
+ *
+ * @author Jörn Friedrich Dreyer <jfd@owncloud.com>
+ * @copyright Jörn Friedrich Dreyer 2015
+ */
+
+namespace OC\User;
+
+class NoUserException extends \Exception {}
```
Comment (blizzz) [inline lib/private/user/nouserexception.php]: blank line missing

**Comment by blizzz** [comment]
The curl test as taken from here sends the error back as HTML instead:

```
$ curl -X PROPFIND -H "Depth: 1" -u stable8:stable8 http://zara.owncloud.bzoc/stable8/remote.php/webdav/  | xmllint --format -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8364    0  8364    0     0  53550      -:63: 0 --:-parser -:-error : Opening and ending tag mismatch: meta line 15 and head
-                                       </head>
                                               ^
100  8364    0  8364    0     0  52785      0 --:--:-- --:--:-- --:--:-- 52603
-:100: parser error : Opening and ending tag mismatch: meta line 14 and html
</html>
       ^
-:101: parser error : Premature end of data in tag head line 9

^
-:101: parser error : Premature end of data in tag html line 7

^
```

```
$ curl -X PROPFIND -H "Depth: 1" -u stable8:stable8 http://zara.owncloud.bzoc/stable8/remote.php/webdav/  
<!DOCTYPE html>
[…]
        <h2><strong>Internal Server Error</strong></h2>
                <p>The server encountered an internal error and was unable to complete your request.</p>
                <p>Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.</p>
                <p>More details can be found in the server log.</p>
        <br />
[…]
```

LDAP was shut down.

**Comment by MorrisJobke** [comment]
> The curl test as taken from here sends the error back as HTML instead:

Yes. This is a bug of sabreDav. Which is fixed in 8.1 (not possible to backport a whole new version of sabre dav to stable8)

Essential here is that the response has a 503 HTTP error code and an invalid WebDAV response body (not a valid body with the share removed, because this implies the client to remove the folder locally)

**Comment by MorrisJobke** [comment]
`curl -v ...` is your friend for the HTTP headers ;)

**Comment by blizzz** [comment]
> curl -v ... is your friend for the HTTP headers ;)

I was too lazy to open up the man page :speak_no_evil:

**Comment by blizzz** [comment]
> Essential here is that the response has a 503 HTTP error code and an invalid WebDAV response body (not a valid body with the share removed, because this implies the client to remove the folder locally)

Nah, it's a 500:

```
< HTTP/1.1 500 Internal Server Error
< Date: Mon, 11 May 2015 11:14:32 GMT
* Server Apache/2.4.10 (Ubuntu) is not blacklisted
< Server: Apache/2.4.10 (Ubuntu)
< Set-Cookie: oc3al6jnhd6p=1kdo5v27s5cfs4p1r4hmfc8m23; path=/stable8; HttpOnly
< Expires: Thu, 19 Nov 1981 08:52:00 GMT
< Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
< Pragma: no-cache
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< X-Frame-Options: Sameorigin
< Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src *; font-src 'self' data:; media-src *; connect-src *
< X-Robots-Tag: none
< Set-Cookie: oc3al6jnhd6p=e48tdvotcicl5vt9cledsi6s75; path=/stable8; HttpOnly
< Connection: close
< Transfer-Encoding: chunked
< Content-Type: text/html; charset=UTF-8
```

**Comment by MorrisJobke** [comment]
Yes. I meant that :see_no_evil:

**Comment by blizzz** [comment]
Splendid.  :+1:

**Pull request merged**